task/oci-copy-oci-ta/0.1/oci-copy-oci-ta.yaml

```diff
@@ -85,16 +90,33 @@ spec:
       volumeMounts:
         - mountPath: /var/lib/containers
           name: varlibcontainers
+      env:
+        - name: BEARER_TOKEN
+          valueFrom:
+            secretKeyRef:
+              key: token
+              name: $(params.BEARER_TOKEN_SECRET_NAME)
+              optional: true
       script: |
-        set -eu
+        set -e
         set -o pipefail
 
+        CURL_ARGS=()
+        if [ -n "${BEARER_TOKEN}" ]; then
+          echo "Found bearer token. Using it for authentication."
+          CURL_ARGS+=(-H "Authorization: Bearer ${BEARER_TOKEN}")
+        else
```

Comment:
It doesn't feel right to send the same Bearer token for all the requests considering that the oci-copy file can reference artifacts coming from multiple servers

If the huggingface token can also be used with HTTP Basic, maybe this task could use the same approach as the prefetch task (a .netrc file). Otherwise, IMO we at least need to document that the same token will be sent to all the servers

Reply:
Or maybe the secret could be in a format like
```
huggingface.co: $token1
example.com: $token2
```

Reply:
I spent some time looking today for basic auth options, and while the [hugging face docs](https://huggingface.co/docs/hub/security-tokens#how-to-use-user-access-tokens) say that "User Access Tokens can be used in place of a password to access the Hugging Face Hub ... with basic authentication.", I couldn't actually get it to work with curl - trying both the `-u username:password` form and the `~/.netrc` file.

I gave up, and proposed documentation in f61a705.

The specially formatted secret could work, but - I'm not sure it is worth it at this point. Also, introducing *two* new file formats for this task feels excessive.

Reply:
Rather than a file format, I was thinking the secret would just contain multiple keys, one per hostname. The task would have to parse the hostname from each url and get the bearer token from `cat /tokens-secret/$hostname`

But documenting the limitations should be good enough for now